build/params_debug.go

```diff
@@ -4,6 +4,9 @@
 package build
 
 func init() {
+	// TODO: This doesn't enable insecure validation in the actors, which is likely a problem.
```

Comment:
> Then should we not be using the special FVM bundle that disables window post checking, etc?

So it breaks down into 2 cases: itests, and testnets. 

For itests, we **should** use the special faked FVM bundle if `mockProofs` is set. We are currently using the special bundle for _all_ itests, because we use the special bundle if InsecurePoStValidation is true, and we set InsecurePoStValidation to true for (almost) all itests. This isn't strictly correct, but...fine.

 For testnets, we should **definitely** _not_ use the special bundle if InsecurePoStValidation isn't set (we currently don't), and we should **probably** not use it even if InsecurePoStValidation is set (we currently do). That's cuz the Lotus miner will compute real Window PoSts even if InsecurePoStValidation is set, and Idk what that will do to the faked proofs bundle. The good news is that's what we currently do (we never use the fake-proofs bundle on testnets).

Reply:
Ok, so, this is all broken.

1. We currently globally set `build.InsecurePoStValidation = true` for all tests. We then try to explicitly set it for some tests, but we've already set it (https://github.com/filecoin-project/lotus/blob/54a409777ba0a58d5c1cfa9468fd7ca4a171bc08/itests/kit/init.go#L23) so that doesn't matter.
2. We don't even look at `mockProofs`.

Reply:
Fixed:

1. I'm no longer explicitly setting InsecurePoStValidation inside itests.
2. I'm deciding which post validation to use based on the mockProofs option, not InsecurePoStValidation.